content/building/build-versioning.md

```diff
@@ -46,3 +46,141 @@ set -x
+
+Add the following script under your `scripts` field for `codemagic.yaml`, or as a custom [Pre-build script](https://docs.codemagic.io/flutter/custom-scripts/) in UI settings (for Flutter projects):
+
+```bash
+export APP_STORE_CONNECT_PRIVATE_KEY=$(echo $APP_STORE_CONNECT_PRIVATE_KEY | base64 --decode) # if you encrypted the file itself, not its content
+LATEST_BUILD_NUMBER=$(app-store-connect get-latest-app-store-build-number '1234567890') # The argument is your application's Apple ID
+agvtool new-version -all $(($LATEST_BUILD_NUMBER + 1))
+```
+
+To use the latest build number from Testflight use a similar script:
+
+```bash
+export APP_STORE_CONNECT_PRIVATE_KEY=$(echo $APP_STORE_CONNECT_PRIVATE_KEY | base64 --decode) # if you encrypted the file itself, not its content
+LATEST_BUILD_NUMBER=$(app-store-connect get-latest-testflight-build-number '1234567890') # The argument is your application's Apple ID
+agvtool new-version -all $(($LATEST_BUILD_NUMBER + 1))
+```
+
+{{<notebox>}}
+You can specify an optional argument `--app-store-version=APP_STORE_VERSION` to get the latest build number for a particular version of your application (`CFBundleShortVersionString`). Check the details for [get-latest-app-store-build-number](https://github.com/codemagic-ci-cd/cli-tools/blob/master/docs/app-store-connect/get-latest-app-store-build-number.md#optional-arguments-for-action-get-latest-app-store-build-number) or for [get-latest-testflight-build-number](https://github.com/codemagic-ci-cd/cli-tools/blob/master/docs/app-store-connect/get-latest-testflight-build-number.md#optional-arguments-for-action-get-latest-testflight-build-number).
+{{</notebox>}}
+
+Alternatively, if you use `YAML` configuration, you may just export the value to an environment variable and use it under your `CFBundleVersion` in `Info.plist`.
+
+## Use Codemagic CLI Tools to get the latest build number from Google Play
+
+Use [get-latest-build-number](https://github.com/codemagic-ci-cd/cli-tools/blob/master/docs/google-play/get-latest-build-number.md#get-latest-build-number) action from [google-play](https://github.com/codemagic-ci-cd/cli-tools/tree/master/docs/google-play#google-play) Codemagic CLI Tool to get the latest build number from Google Play.
+
+In order to do that, you need to provide API access to Google Play Developer API by providing `GCLOUD_SERVICE_ACCOUNT_CREDENTIALS` as arguments to the action, as defined below.
+
+Additionally, you will need to provide the package name of the app in Google Play Console (Ex. `com.google.example`).
+
+### Creating Google service account credentials
+
+You will need to set up a service account in Google Play Console and create a JSON key with credentials. See how to [set up a service account and create a key](../knowledge-base/google-play-api/#setting-up-the-service-account-on-google-play-and-google-cloud-platform).
+
+### Saving the API access argument to environment variables in `codemagic.yaml` config
```

Comment:
Here too a note is needed that if they have already configured this in yaml, then they're good to go

Reply:
added it [here](https://github.com/codemagic-ci-cd/codemagic-docs/pull/267/files#diff-f6e40a5795c4d7de47d17ed5b2850ac8a06b55f09f22d3bf4611e3a4bfbffe4bR145)